src/Infrastructure.EntityFramework/Repositories/OrganizationRepository.cs

```diff
@@ -139,4 +141,41 @@ public override async Task DeleteAsync(Core.Entities.Organization organization)
             await dbContext.SaveChangesAsync();
         }
     }
+
+    public async Task<Core.Entities.Organization> GetByLicenseKeyAsync(string licenseKey)
+    {
+        using (var scope = ServiceScopeFactory.CreateScope())
+        {
+            var dbContext = GetDatabaseContext(scope);
+            var organization = await GetDbSet(dbContext)
+                .Where(o => o.LicenseKey == licenseKey)
+                .FirstOrDefaultAsync();
```

Comment:
This query can be reduced:
```suggestion
            var organization = await GetDbSet(dbContext)
                .FirstOrDefaultAsync(o => o.LicenseKey == licenseKey);
```

Reply:
Nice one! Didn't know it could take a predicate.